### PR TITLE
Replace instances of SHA-1 usage with SHA-256.

### PR DESCRIPTION
**Pull request by ckelleyRH**

### CODE REVIEW

File: base/ca/src/main/java/org/dogtagpki/ct/CTEngine.java [hunk 1]
```diff
@@ -590,7 +590,7 @@ boolean verifySCT(CTResponse response, byte[] tbsCert, String logPublicKey, ICer
      *  enum HashAlgorithm {
      *      none, md5, sha1, sha224, sha256, sha384, sha512};
      */
-    enum HashAlgorithm {none, MD5, SHA1, SHA224, SHA256, SHA384, SHA512};
+    enum HashAlgorithm {none, MD5,SHA224, SHA256, SHA384, SHA512};
```
Comment (edewata) [inline base/ca/src/main/java/org/dogtagpki/ct/CTEngine.java:593]: Needs a space after comma :)

**Review by edewata — COMMENTED** [review]
Just one minor comment, but like in the other SHA-1 PRs, I'd have to defer to @ladycfu and @jmagne.

File: base/ca/src/main/java/org/dogtagpki/ct/CTEngine.java (hunk 1 above)
Comment (c-dorney) [reply base/ca/src/main/java/org/dogtagpki/ct/CTEngine.java:593]: Very pedantic Endi.. but I agree :joy: 
```suggestion
    enum HashAlgorithm {none, MD5, SHA224, SHA256, SHA384, SHA512};
```

Comment (ckelleyRH) [reply base/ca/src/main/java/org/dogtagpki/ct/CTEngine.java:593]: I apologise profusely for this scandalous oversight, you have actually prevented a bug with this. :rofl: 

I realised that I hadn't updated the comment directly above, which led me to read the code again more thoroughly. We define this enum, but then we only actually return not-`null` if we hit the specific ordinal value of `SHA-256`, by removing `SHA-1` we would have only returned not-`null` for `SHA-384`, which could have been not good.

**Pull request closed without merging**